services/QuillLMS/app/controllers/students_controller.rb

```diff
@@ -37,6 +37,37 @@ def update_email
     end
   end
 
+  def join_classroom
+    if current_user
+      if current_user.role === 'student'
+        classcode = params[:classcode].downcase
+        begin
+          classroom = Classroom.find_by(code: classcode)
+          Associators::StudentsToClassrooms.run(current_user, classroom)
+          JoinClassroomWorker.perform_async(current_user.id)
+        rescue NoMethodError => exception
+          if Classroom.unscoped.find_by(code: classcode).nil?
+            InvalidClasscodeWorker.perform_async(current_user.id, params[:classcode], classcode)
+            flash[:error] = "Oops! There is no class with the code #{classcode}. Ask your teacher for help."
+          else
+            flash[:error] = "Oops! The class with the code #{classcode} is archived. Ask your teacher for help."
+          end
+          flash.keep(:error)
+          redirect_to '/profile'
+        else
+          redirect_to "/classrooms/#{classroom.id}?joined=success"
+        end
+      else
+        flash[:error] = 'Oops! That link is only accessible for students.'
+        redirect_to '/profile'
+      end
+    else
+      session[:post_auth_redirect] = request.env['PATH_INFO']
```

Comment:
because the sessions controller uses `:post_auth_redirect` but the accounts controller uses `:post_sign_up_redirect` and I didn't want to add a third variable to check for both of them.

Reply:
Some day we should probably consolidate to a single value, but this works just fine for now.